test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageSourceMappingProviderTests.cs

```diff
@@ -410,5 +410,242 @@ public void AddOrUpdatePackageSourceMapping_WithAClearItem_WithANewPattern_AddIn
+    </packageSourceMapping>
+</configuration>";
+
+            result.Replace("\r\n", "\n")
+                .Should().BeEquivalentTo(
+                File.ReadAllText(configPath1).Replace("\r\n", "\n"));
+        }
+
+        [Fact]
+        public void SavePackageSourceMappings_WithMappingsAddedRemovedUpdated_AddRemoveUpdate()
+        {
+            // Arrange
+            using var mockBaseDirectory = TestDirectory.Create();
+            var configPath1 = Path.Combine(mockBaseDirectory, "NuGet.Config");
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSourceMapping>
+        <clear />
+        <packageSource key=""nuget.org"">
+            <package pattern=""stuff"" />
+        </packageSource>
+    </packageSourceMapping>
+</configuration>");
+
+            PackagePatternItem testPackagePatternItem = new PackagePatternItem("stuff");
+            List<PackagePatternItem> testPackagePatternItems = new List<PackagePatternItem>();
+            testPackagePatternItems.Add(testPackagePatternItem);
+            PackageSourceMappingSourceItem testMappingItem = new PackageSourceMappingSourceItem("nuget.org", testPackagePatternItems);
+            PackageSourceMappingSourceItem testMappingItem2 = new PackageSourceMappingSourceItem("nuget2.org", testPackagePatternItems);
+            List<PackageSourceMappingSourceItem> tempMappings = new List<PackageSourceMappingSourceItem>();
+            tempMappings.Add(testMappingItem);
+            tempMappings.Add(testMappingItem2);
+
+            var settings = Settings.LoadSettingsGivenConfigPaths(new string[] { configPath1 });
+
+            // Act & Assert
+            var sourceMappingProvider = new PackageSourceMappingProvider(settings);
+            sourceMappingProvider.SavePackageSourceMappings(tempMappings);
+
+            //Update
+            List<PackagePatternItem> packagePatternItems = new List<PackagePatternItem>();
+            for (int i = 0; i < 3; i++)
+            {
+                PackagePatternItem tempPackagePatternItem = new PackagePatternItem(i.ToString());
+                packagePatternItems.Add(tempPackagePatternItem);
+            }
+            PackageSourceMappingSourceItem tempMapping = new PackageSourceMappingSourceItem("nuget.org", packagePatternItems);
+            tempMappings.Add(tempMapping);
+
+            //Add
+            PackagePatternItem packagePatternItemAdd = new PackagePatternItem("stuff");
+            List<PackagePatternItem> packagePatternItemsAdd = new List<PackagePatternItem>();
+            packagePatternItemsAdd.Add(packagePatternItemAdd);
+            PackageSourceMappingSourceItem testMappingItemAdd = new PackageSourceMappingSourceItem("newSource", packagePatternItemsAdd);
+
+            tempMappings.Add(testMappingItemAdd);
+            tempMappings.Remove(testMappingItem2);
+
+            sourceMappingProvider.SavePackageSourceMappings(tempMappings);
```

Comment:
Would this call be the 'act' part of the test?

```suggestion
            // Act
            sourceMappingProvider.SavePackageSourceMappings(tempMappings);
```

Reply:
Yeah there is a comment on line 602 labeling it "Act & Assert". That is how the other tests in this file seem to be set up, so I just copied how everything else was labeled.